pkg/query/build_tree.go

```diff
@@ -85,6 +84,9 @@ func (b *iteratorBuilder) buildIteratorFromOperation(p *schema.Permission, op sc
 		return union, nil
 
 	case *schema.RelationReference:
+		if perm.RelationName == "_nil" {
+			return NewEmptyFixedIterator(), nil
```

Comment:
Why is this branch needed?

Reply:
It's an addition for handling the `nil` keyword that, while handled in the schema package, was ignored here (and so the fix snuck in while I was looking at some of the other tests)